src/backend/distributed/worker/worker_partition_protocol.c

```diff
@@ -942,6 +942,12 @@ FilterAndPartitionTable(const char *filterQuery,
 	while (SPI_processed > 0)
 	{
 		int rowIndex = 0;
+
+		if (fileCount == 0)
```

Comment:
Deferred error until here so that we don't mind a lack of shards if there's no results